src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -369,10 +369,17 @@ def _chain_library_resolvers(library_resolvers: list[BaseLibraryResolver]) -> Ba
 
     @staticmethod
     def _chain_import_resolvers(import_resolvers: list[BaseImportResolver]) -> BaseImportResolver:
+        # check that WhiteListResolver comes first when resolving import
```

Comment:
we currently have:
<img width="855" alt="image" src="https://github.com/databrickslabs/ucx/assets/259697/0a516780-e6d1-44ce-bb86-1a59c42733be">

```mermaid
graph TD

PathLookup -->|inject into| resolve_import
DependencyResolver -->|call|resolve_import

resolve_import -.->|method of| BaseImportResolver
BaseImportResolver -.->|subclass| LocalFileResolver
BaseImportResolver -.->|subclass| WhitelistResolver
BaseImportResolver -.->|subclass| StubImportResolver
LocalFileResolver -.->|next| WhitelistResolver
WhitelistResolver -.->|next| StubImportResolver

Whitelist --> WhitelistResolver
FileLoader --> LocalFileResolver
```
this is hard to work with and requires code like this to fix flaw on the higher level.

# Simplify

```mermaid
graph TD

PathLookup -->|inject into| resolve_import
DependencyResolver -->|call|resolve_import
resolve_import -.->|method of| ImportResolver
Whitelist -->|inject into constructor| ImportResolver
FileLoader --> |inject into constructor| ImportResolver
```